### PR TITLE
Add Ubuntu and Azure Linux 3 to Proxy Agent Extension E2E Test and Update Logging 

### PR DESCRIPTION
**Pull request by shahneerali**

### CODE REVIEW

File: proxy_agent_extension/src/linux.rs [hunk 1]
```diff
@@ -5,14 +5,28 @@ use std::thread;
...
 use crate::service_main;
+use proxy_agent_shared::{misc_helpers, telemetry};
 
 #[cfg(not(windows))]
 pub fn start_service_wait() {
     // start service
     let service_state = service_main::service_state::ServiceState::new();
     service_main::run(service_state);
-
+    let message = format!(
```
Comment (ZhidongPeng) [inline proxy_agent_extension/src/linux.rs]: already logged in service_main.rs and proxy_agent_extension\src\handler_main.rs file

File: proxy_agent_extension/src/linux.rs [hunk 2]
```diff
@@ -5,14 +5,28 @@ use std::thread;
...
+        "==============  GuestProxyAgentExtension Start Service, Version: {}, OS Arch: {}, OS Version: {}",
+        misc_helpers::get_current_version(),
+        misc_helpers::get_processor_arch(),
+        misc_helpers::get_long_os_version()
+    );
+    telemetry::event_logger::write_event(
```
Comment (ZhidongPeng) [inline proxy_agent_extension/src/linux.rs]: write file log is good enough, do not use telemetry log.

File: proxy_agent_shared/src/linux.rs [hunk 3]
```diff
@@ -46,6 +47,11 @@ pub fn get_os_version() -> String {
                 }
             }
             Err(_e) => {
+                let message = format!(
+                    "Failed to read os-release file in get_os_version(): {}",
+                    OS_RELEASE_PATH
+                );
+                logger_manager::write_info(message);
```
Comment (ZhidongPeng) [inline proxy_agent_shared/src/linux.rs]: write_warn at least, same as next.

**Review by ZhidongPeng — DISMISSED** [review]

File: proxy_agent_extension/src/linux.rs (hunk 1 above)
Comment (shahneerali) [reply proxy_agent_extension/src/linux.rs]: reverted to original since this is redundant logging

File: proxy_agent_shared/src/linux.rs [hunk 4]
```diff
@@ -46,6 +47,11 @@ pub fn get_os_version() -> String {
                 }
             }
             Err(_e) => {
+                let message = format!(
+                    "Failed to read os-release file in get_os_version(): {}",
+                    OS_RELEASE_PATH
```
Comment (ZhidongPeng) [inline proxy_agent_shared/src/linux.rs]: please log the error to file too, we want more to know why failed.

**Review by ZhidongPeng — DISMISSED** [review]

**Review by ZhidongPeng — APPROVED** [review]

**Pull request merged**